test/extension/numeric/kernel.cpp

```diff
@@ -156,6 +156,31 @@ void test_kernel_1d_reverse_kernel()
     BOOST_TEST_EQ(k.size(), d.size());
 }
 
+void test_kernel_2d_reverse_kernel()
+{
+    std::vector<int> data = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+    gil::detail::kernel_2d<int> d(data.begin(), data.size(), 2, 0);
+
+    BOOST_TEST_EQ(d.center_x(), 0);
+    BOOST_TEST_EQ(d.center_y(), 2);
+
+    auto k = gil::detail::reverse_kernel(d);
+
+    BOOST_TEST_EQ(k.center_x(), d.lower_size());
+    BOOST_TEST_EQ(k.center_x(), 0);
+
+    BOOST_TEST_EQ(k.center_y(), d.right_size());
+    BOOST_TEST_EQ(k.center_y(), 2);
+
+    // std::vector interface
+    BOOST_TEST_EQ(k.size(), d.size());
+
+    for (size_t i = 0; i < k.size() * k.size(); i++)
+    {
+        BOOST_TEST_EQ(d[i], k[(k.size()*k.size()) - 1 - i]);
```

Comment:
Hopefully this does not trigger compiler warning

Reply:
Maybe it will, i overlooked it. Will check and correct it if needed in the morning... :)

Reply:
does not seem to trigger any compiler warning :)

Reply:
👍